app/services/solr_document_builder.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class SolrDocumentBuilder
+  class Error < StandardError; end
+
+  def self.call(resource:, schema: nil)
+    return {} if resource.nil?
+
+    new(resource, schema).document
+  end
+
+  attr_reader :schema, :resource
+
+  def initialize(resource, schema)
+    raise Error, "#{resource.class} has no uuid defined" unless resource.respond_to?(:uuid)
```

Comment:
Personally I like `ArgumentError` for this kind of stuff

Reply:
@rschenk are you saying just use `ArgumentError` or `SolrDocumentBuilder::Error` extends argument error?

Reply:
When I initially wrote that comment, I assumed `SolrDocumentBuilder::Error` was used in more places than just here, although I see now that it's not. The reason why I personally like `ArgumentError` is because it's quite specific so it makes it easy to rescue from, or track down the problem in a stack trace. I had imagined that `SolrDocumentBuilder::Error` would be used other places, like if a schema couldn't be built or something. I guess I felt like the situations where you would have an incorrect argument would be encountered by developers when developing, whereas situations raising `SolrDocumentBuilder::Error` might happen at runtime, when the correct type of object was given to the class but for some reason it failed. I guess it's really just a personal thing.